src/Tree.js

```diff
@@ -75,16 +75,30 @@ Tree.prototype = {
 }
 
 function Node (value, parent) {
-  let node = { depth: 0, parent, value, children: [] }
+  this.parent = parent
+  this.value  = value
 
   if (parent) {
-    parent.children.push(node)
-    parent.next = node
+    this.depth   = parent.depth + 1
+    this.sibling = parent.next
 
-    node.depth = parent.depth + 1
+    parent.next = this
   }
+}
+
+Node.prototype = {
+  depth: 0,
+
+  get children() {
```

Comment:
This will require IE9+ Browser support. We only use this for dev tools. What do you think?